rxnetty/src/main/java/io/reactivex/netty/client/ConnectionPoolImpl.java

```diff
@@ -54,7 +58,13 @@
     private final RxClient.ServerInfo serverInfo;
     private final PoolConfig poolConfig;
     private final ScheduledExecutorService cleanupScheduler;
-    private final AtomicBoolean isShutdown = new AtomicBoolean();
+    private final AtomicInteger aquiredConnectionsCounter = new AtomicInteger();
+    //is needed to not accept any new connections that are aquired
+    private final AtomicBoolean isShutdownRequested = new AtomicBoolean();
+    //is needed to not perform the shutdown twice
+    private final AtomicBoolean isShutdownPerformed = new AtomicBoolean();
+    //is needed to not perform shutdown during the aquire of a connection
+    private final ReadWriteLock shutdownLock = new ReentrantReadWriteLock();
```

Comment:
I kind of like your approach if it was without the lock :) I somehow don't like the idea of locking for a non-critical operation. Having said that, it may be clearer to me if I see the code.

So, there isn't a connection leak but a discrepancy in metrics, rite?